internal/rbd/rbd_util.go

```diff
@@ -786,24 +786,18 @@ func createSnapshot(ctx context.Context, vol *rbdVolume, pOpts *rbdSnapshot, cr
 	return err
 }
 
-func deleteSnapshot(ctx context.Context, pOpts *rbdSnapshot, cr *util.Credentials) error {
-	var output []byte
-
-	klog.V(4).Infof(util.Log(ctx, "rbd: snap rm %s using mon %s"), pOpts, pOpts.Monitors)
-	args := []string{"snap", "rm", pOpts.String(), "--id", cr.ID, "-m", pOpts.Monitors, "--keyfile=" + cr.KeyFile}
-
-	output, err := execCommand("rbd", args)
+func deleteSnapshot(ctx context.Context, vol *rbdVolume, pOpts *rbdSnapshot, cr *util.Credentials) error {
+	klog.V(4).Infof(util.Log(ctx, "rbd: snap rm %s/%s@%s using mon %s"), pOpts.Pool, pOpts.RbdImageName, pOpts.RbdSnapName, pOpts.Monitors)
 
+	image, err := vol.open()
 	if err != nil {
-		return errors.Wrapf(err, "failed to delete snapshot, command output: %s", string(output))
-	}
-
-	if err := undoSnapReservation(ctx, pOpts, cr); err != nil {
-		klog.Errorf(util.Log(ctx, "failed to remove reservation for snapname (%s) with backing snap (%s): %s"),
-			pOpts.RequestName, pOpts, err)
+		return err
 	}
+	defer image.Close()
 
-	return nil
+	snap := image.GetSnapshot(pOpts.RbdSnapName)
```

Comment:
please check if `snap` is `nil`

Reply:
no idea in which case it can return `nil` as per https://github.com/ceph/go-ceph/blob/v0.2.0/rbd/rbd.go#L958. adding nil check for safer side.